### DISCUSSION TITLE
Parameter extraction from [CASEDATA] (fraction, sqrt, ...)

### DISCUSSION THREAD

**Discussion by turbulentphil** [Nek5000/nekRS#592] [Q&A] [answered]
Dear Neks,

I would like to define some control parameters in the `[CASEDATA]` section of the _.par_ file. Since other values -- such as `viscosity` and `conductivity` (see the `rbc` example) -- allow for the input of simple formulas (i.e., taking square roots or computing fractions), I expected this to work also for `[CASEDATA]` parameters. However, this seems not to work for `[CASEDATA]` parameters.

Here is an example:
1. My parameter is defined in the .par file under `[CASEDATA]` via `R_0 = sqrt(1.0f)`
2. The parameter is assigned in the `UDF_Setup0` function via `platform->par->extract("casedata", "R_0", R_0);` given `static dfloat R_0;`
3. The read parameter is checked by printing the value via `std::cout << "   R_0 = " << R_0 << std::endl;`
4. The log file contains `R_0 = 0`
Similarly, a simple fraction `R_0 = (1.0f/80.0f)` results in a zero value in the output too.

Here is my question:
Is there any way to read my own parameters in a similarly sophisticated fashion like other standard parameters, such as `viscosity` and `conductivity` (i.e., support fractions etc.)? 

Cheers, Philipp

**Comment by stgeke** [comment, marked as answer]
Please be aware that the formula parser is restricted to non-user-defined sections, as we need to know a priori which values are strings and which are formulas. While it is possible to implement logic to determine this, such functionality is not currently available.

**Reply by turbulentphil** [reply]
I see, thanks for your answer!

**Reply by stgeke** [reply]
Will make the parser available in the UDF. Then you can evaluate anything you want.
